WordPressKit/DomainsServiceRemote.swift

```diff
@@ -130,12 +130,16 @@ public class DomainsServiceRemote: ServiceRemoteWordPressComREST {
     }
 )
     */
-    public func getDomainSuggestions(base query: String, success: @escaping ([String]) -> Void, failure: @escaping (Error) -> Void) {
+    public func getDomainSuggestions(base query: String,
+                                     includeWordPressDotCom: Bool = true,
+                                     onlyWordPressDotCom: Bool = true,
+                                     success: @escaping ([String]) -> Void,
+                                     failure: @escaping (Error) -> Void) {
         let endPoint = "domains/suggestions"
         let servicePath = path(forEndpoint: endPoint, withVersion: ._1_1)
         let parameters: [String: AnyObject] = ["query": query as AnyObject,
-                                               "include_wordpressdotcom": true as AnyObject,
-                                               "only_wordpressdotcom": true as AnyObject]
+                                               "include_wordpressdotcom": includeWordPressDotCom as AnyObject,
```

Comment:
I might be misunderstanding how this API is supposed to work server-side, but I don't think modeling those options as two `Bool`s is the best choice here. Not sure how the backend would behave, but this API as-is allows you to pass `includeWPCom` as `false` and `onlyWPCom` as `true`, which, logically, should always return empty set :) 

What about a three state enum, like 
```
enum TooTiredToFigureOutASmartNamePleaseForgiveMe {
    case noWordpressDotCom
    case includeWordPressDotCom
    case onlyWordPressDotCom
}
``` 
or something similar?

Reply:
oh yes you are probably right but It looks like it is something that needs to be fixed on the API side first(I mean the serverside), isn't it?

Reply:
Hmm, not really, I think we can model this "correctly" on our end only. If you take the example enum I posted above, you can do something like that:

```swift
let parameters: [String: AnyObject]
switch includeParameter {
     case .noWordPressDotCom: parameters = ["include_wordpressdotcom": false]
     case .includeWordPressDotCom: parameters  = ["include_wordpressdotcom" true, "only_wordpressdotcom": false]
     case .onlyWordPressDotCom: parameters =  ["include_wordpressdotcom" true, "only_wordpressdotcom": true]
}
```

(or something to that effect)

We'd still get a sensible API for this on our end, and the backend can keep it's own weirdness ;P

Reply:
Ok sure I'll update it on our side. I am just trying to understand how things are moving forward between back end and client teams

Reply:
I've updated this